frame/system/src/lib.rs

```diff
@@ -1316,6 +1317,8 @@ impl<T: Config> Pallet<T> {
 		// populate environment
 		ExecutionPhase::<T>::put(Phase::Initialization);
 		storage::unhashed::put(well_known_keys::EXTRINSIC_INDEX, &0u32);
+		let entropy = (b"frame_system::initialize", parent_hash).using_encoded(blake2_256);
+		storage::unhashed::put(well_known_keys::INTRABLOCK_ENTROPY, &entropy[..]);
```

Comment:
```suggestion
		storage::unhashed::put_raw(well_known_keys::INTRABLOCK_ENTROPY, &entropy[..]);
```

Otherwise `put` encodes again and pretends the length, so it looses one byte of entropy when only reading the first 32.

Reply:
Linking to PR addressing this: https://github.com/paritytech/substrate/pull/14152

Reply:
fixed lengths arrays are not prepended with its length.

Reply:
It is a slice though.

Reply:
Good point :see_no_evil: